docs/knowledgebase/runtime/macros.md

```diff
@@ -189,6 +202,217 @@ also inserts lifecycle trait implementations for `Module`, e.g.
+
+For example: 
+```rust
+#[pallet::event]
+#[pallet::metadata(u32 = "SpecialU32")]
+pub enum Event<T: Config> {
+	Proposed(u32, T::AccountId),
+}
+```
+
+**Docs**
+- See [documentation](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#event-palletevent-optional)
+### #[pallet::storage]
+
+**When to Use** 
+
+Optionally, as it can be used multiple times.
+
+**What it Does** 
+
+To define some abstract storage inside runtime storage.
+
+**Note:** this changes how storage is declared compared to v1. Here, `[pallet::storage]` is using Rust like storage maps to do storage, such that it can be generic over the pallet. 
+
+**Docs**
+- See [documentation](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#storage-palletstorage-optional)
+- See Rust docs for [StorageMap](https://docs.rs/storage-map/0.1.2/storage_map/struct.StorageMap.html) and [HashMap syntax](https://doc.rust-lang.org/book/ch08-03-hash-maps.html)
+
+### Others
+
+Other FRAME v2 macro attributes include:
+- [#[pallet::genesis_config]](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#genesis-config-palletgenesis_config-optional) which used to be in `decl_storage` in v1. This gives runtime engineers more flexibility to define `genesis_config`.
+- [#[pallet::type_value]](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#type-value-pallettype_value-optional)
+- [#[pallet::genesis_build]](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#genesis-build-palletgenesis_build-optional)
+- [#[pallet::inherent]](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#inherent-palletinherent-optional)
+- [#[pallet::validate_unsigned]](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#validate-unsigned-palletvalidate_unsigned-optional)
+- [#[pallet::origin]](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#origin-palletorigin-optional)
+
+#### FRAME v1 vs. v2
+Things to note:
+- Using instantiable pallets with v2 requires a different approach than for v1. See the guide [here](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#example-for-pallet-without-instance) for more information on how that works.
+
+- In v2, `pub(super)` is used to make things private to the pallet but accessible to the associated namespace. 
+
+- v2 is backwards compatible with v1, however v1 will cease getting improvement features. Storage will remain the same.
+
+- Use [subsee](https://github.com/ascjones/subsee) as a tool to query metadata. 
+
+- `RawEvent` is no longer used in v2, all instances of events are declred as `Event`. 
+
+Here's a brief comparative overview:
+- v2 leans more heavily on Rust to define types and uses crates by writing code within the constaints of the macro being used, whereas v1 relies on runtime engineers to write more code to declare traits and types
+- v2 macros removes the need to write attributes 
```

Comment:
This is a mistake &mdash; it's meant to read "_v2 macros minimize the amount of logic developers need to write that is not pertinent to their pallet's logic_" but it's really just repeating the first point. Is there anything you would add to this list?

Reply:
From my point of view v2 is structure to be more rust idiomatic, almost like a derive macro would do.

The input is a valid rust syntax, the macro check it is declared correctly (e.g. the `Config` trait has supertrait `frame_system::Config`), slightly modify this input (e.g. replacement of some `_` ident in pallet::storage and pallet::pallet items) + generate new items (e.g. the `Call` enum).

Whereas v1 was a non rust syntax and more like a DSL. This makes another DSL to learn in order to understand how pallet works. The specific structure of input in macro made it difficult to understand what the macro was doing and how it can be used.